Asthma/TabBarControllers/Dashboard/APHCalendarDataModel.m

```diff
@@ -194,128 +228,85 @@ -(NSDictionary *)userCompliedWithDailyScheduledTasks:(APHCalendarTaskType)task i
+#pragma mark - Fetch Requests
+- (NSArray *)dailySurveyScheduledTasksForDateRange: (APCDateRange *)dateRange
 {
-    NSFetchRequest * request = [APCScheduledTask request];
-    request.shouldRefreshRefetchedObjects = YES;
-    NSPredicate * datePredicate = [NSPredicate predicateWithFormat:@"(startOn >= %@) AND (endOn <= %@) AND task.taskID == %@", dateRange.startDate, dateRange.endDate, kDailySurveyTaskID];
+    NSArray *dailyScheduledTasks = nil;
+    APCAppDelegate *appDelegate = (APCAppDelegate *)[[UIApplication sharedApplication] delegate];
+    NSSortDescriptor *sortDescriptor = [[NSSortDescriptor alloc] initWithKey:@"startOn" ascending:NO];
+    NSFetchRequest *request = [APCScheduledTask request];
+    [request setShouldRefreshRefetchedObjects:YES];
+    NSDate *startDate = dateRange.startDate;
+    NSDate *endDate = dateRange.endDate;
     
-    NSPredicate * completionPredicate = nil;
-    if (completed != nil) {
-        completionPredicate = [completed isEqualToNumber:@YES] ? [NSPredicate predicateWithFormat:@"completed == %@", completed] :[NSPredicate predicateWithFormat:@"completed == nil ||  completed == %@", completed] ;
-    }
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"(startOn >= %@) AND (startOn <= %@) AND task.taskID == %@", startDate, endDate, kDailySurveyTaskID];
+    request.predicate = predicate;
+    request.sortDescriptors = @[sortDescriptor];
     
-    NSPredicate * finalPredicate = completionPredicate ? [NSCompoundPredicate andPredicateWithSubpredicates:@[datePredicate, completionPredicate]] : datePredicate;
-    request.predicate = finalPredicate;
+    NSError *error = nil;
+    dailyScheduledTasks = [appDelegate.dataSubstrate.mainContext executeFetchRequest:request error:&error];
+    APCLogError2(error);
     
-    NSSortDescriptor *titleSortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"task.taskTitle" ascending:YES];
-    NSSortDescriptor * completedSortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"completed" ascending:YES];
-    request.sortDescriptors = @[completedSortDescriptor, titleSortDescriptor];
-    
-    NSError * error;
-    NSArray * array = [context executeFetchRequest:request error:&error];
-    if (array == nil) {
-        APCLogError2 (error);
-    }
-    
-    NSMutableArray * filteredArray = [NSMutableArray array];
-    
-    for (APCScheduledTask * scheduledTask in array) {
-        if ([scheduledTask.dateRange compare:dateRange] != kAPCDateRangeComparisonOutOfRange) {
-            [filteredArray addObject:scheduledTask];
-        }
-    }
-    return filteredArray.count ? filteredArray : nil;
+    return dailyScheduledTasks;
 }
 
-+ (NSArray *)allScheduledWeeklyPromptTasksForDateRange: (APCDateRange*) dateRange completed: (NSNumber*) completed inContext: (NSManagedObjectContext*) context
+- (NSArray *)weeklySurveyScheduledTasksForDateRange: (APCDateRange *)dateRange
```

Comment:
Can this be consolidated with `dailySurveyScheduledTasksForDateRange:`?
I noticed the only difference is `task.taskID`.